git-sync/main.go

```diff
@@ -213,3 +227,30 @@ func setupGitAuth(username, password, gitURL string) error {
 
 	return nil
 }
+
+func setupGitSSH() error {
+	log.Println("setting up git SSH credentials")
+
+	if _, err := os.Stat("/etc/git-secret/id-rsa"); err != nil {
+		return fmt.Errorf("error: could not find SSH secret: %v", err)
+	}
+
+	// Secret permissions must be restricted to use it as SSH key
+	if err := os.Chmod("/etc/git-secret/id-rsa", 0400); err != nil {
+
+		// Need to make sure secret volume isn't mounted readOnly, which gives default permission 0444
+		// 0444 is not restrictive enough to use as SSH key, but also prevents chmod
+		return fmt.Errorf("error running chmod on secret (make sure secret volume is NOT mounted with readOnly=true): %v", err)
+	}
+
+	wrapperScript := []byte(SSHWrapper)
+	if err := ioutil.WriteFile("/tmp/SSHWrapper.sh", wrapperScript, 0744); err != nil {
```

Comment:
why not just put this into the container image as part of the dockerfile instead of doing all this?